rclpy/rclpy/client.py

```diff
@@ -55,8 +55,6 @@ def unblock(future):
         future.add_done_callback(unblock)
 
         event.wait()
-        if future.exception() is not None:
-            raise future.exception()
```

Comment:
With the change to `result()` this is not necessary anymore since it happens implicitly.